python/ray/data/tests/test_dataset.py

```diff
@@ -1487,6 +1487,15 @@ def test_schema_lazy(ray_start_regular_shared):
     assert ds._plan.execute()._num_computed() == 1
 
 
+def test_count_lazy(ray_start_regular_shared):
+    ds = ray.data.range(100, parallelism=10)
+    # We do not kick off the read task by default.
+    assert ds._plan._in_blocks._num_computed() == 0
+    assert ds.count() == 100
+    # Getting number of rows should not trigger execution of any read tasks.
```

Comment:
Is this always true for other ways of dataset creation, like read_csv?

Reply:
yeah I think so, they all use same code path - `read_datasource()`.

Reply:
Updated comment.